File: src/main/java/io/bindernews/thegrackle/cards/TargetingComputer.java
package io.bindernews.thegrackle.cards;

import com.megacrit.cardcrawl.actions.common.ApplyPowerAction;
import com.megacrit.cardcrawl.core.AbstractCreature;
import com.megacrit.cardcrawl.powers.watcher.VigorPower;
import io.bindernews.bnsts.CardVariables;

public class TargetingComputer extends BaseCard {
    public static final CardConfig C =
            new CardConfig("TargetingComputer", CardType.POWER, CardRarity.COMMON, CardTarget.SELF);
    static final CardVariables VARS = CardVariables.config(c -> {
        c.cost(1, -1);
        c.magic(4, 8);
    });

    public TargetingComputer() {
        super(C, VARS);
    }

    @Override
    public void apply(AbstractCreature p, AbstractCreature m) {
        addToBot(new ApplyPowerAction(p, p, new VigorPower(p, magicNumber), magicNumber));
    }
}

File: src/main/kotlin/io/bindernews/thegrackle/cards/TargetingComputer.kt
package io.bindernews.thegrackle.cards

import com.megacrit.cardcrawl.actions.common.ApplyPowerAction
import com.megacrit.cardcrawl.core.AbstractCreature
import com.megacrit.cardcrawl.powers.watcher.VigorPower
import io.bindernews.bnsts.CardVariables

class TargetingComputer : BaseCard(C, VARS) {
    override fun apply(p: AbstractCreature, m: AbstractCreature) {
        addToBot(ApplyPowerAction(p, p, VigorPower(p, magicNumber), magicNumber))
    }

    companion object {
        @JvmField val C = CardConfig("TargetingComputer", CardType.SKILL, CardRarity.COMMON, CardTarget.SELF)
        val VARS = CardVariables.config { c ->
            c.cost(1)
            c.magic(3, 6)
        }
    }
}